worker/uniter/paths.go

```diff
@@ -125,14 +127,36 @@ func NewWorkerPaths(dataDir string, unitTag names.UnitTag, worker string, isRemo
 	stateDir := join(baseDir, "state")
 
 	newSocket := func(name string, abstract bool) sockets.Socket {
+		// IP address to use for the socket can either be an env var
+		// (when we are the caas operator and are creating the socket to listen),
+		// or inside a YAML file (when we are juju-run and need to see where
+		// to connect to).
 		podIP := os.Getenv(provider.OperatorPodIPEnvName)
+		if podIP == "" {
```

Comment:
shouldn't `OperatorPodIPEnvName` always non-empty?

Reply:
not when this method is called from juju-run

Reply:
OperatorPodIPEnvName/JUJU_OPERATOR_POD_IP only seems to be injected into the operator pod. Regardless I'll be adding TLS certs/authentication information into operator.yaml, I think I'd prefer just one mechanism for specifying the dial-back information juju-run and actions.